usr.bin/renice/renice.c

```diff
@@ -126,24 +127,72 @@ static int
 donice(int which, int who, int prio, bool incr)
 {
 	int oldprio;
+	const char *who_type;
+	char who_str[20];
 
+	/* Determine the type of identifier we're working with */
+	switch (which) {
+	case PRIO_PROCESS:
+		who_type = "process";
+		break;
+	case PRIO_PGRP:
+		who_type = "process group";
+		break;
+	case PRIO_USER:
+		who_type = "user";
+		break;
+	default:
+		who_type = "unknown";
+		break;
+	}
+
+	/* Get current priority */
 	errno = 0;
 	oldprio = getpriority(which, who);
-	if (oldprio == -1 && errno) {
-		warn("%d: getpriority", who);
-		return (1);
+	if (oldprio == -1) {
+		if (errno == EPERM) {
+			fprintf(stderr, "Permission denied: cannot get priority for %s %d\n",
+				who_type, who);
+			return (1);
+		} else if (errno == ESRCH) {
+			fprintf(stderr, "%s %d not found\n", who_type, who);
+			return (1);
+		} else {
+			warn("%s %d: getpriority", who_type, who);
```

Comment:
since we include the error message with warn here that's substantially similar to the fpritnfs above, how is this better?